### PR TITLE
Rework the Azure build system

### PR DESCRIPTION
**Pull request by scholzj**
This PR reworks the Azure pipelines:
* Uses separate stages and jobs and shares artifacts between them
* Adds new release pipeline to facilitate proper release process
* Adds new CVE re-spin pipeline to allow us to re-spin images in case of case image CVEs

This follows what we already use for the Drain Cleaner and what was discussed and agreed on during the Community calls.

### CODE REVIEW

**Comment by scholzj** [comment]
This should be read for review now.

File: .azure/templates/jobs/build_docs.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      - template: '../steps/setup_java.yaml'
+        parameters:
+          JDK_PATH: '/usr/lib/jvm/java-11-openjdk-amd64'
+          JDK_VERSION: '11'
+      - template: '../steps/setup_asciidoc.yaml'
+#            - bash: "make docu_check"
+#              displayName: "Docu checks"
+#              env:
+#                MVN_ARGS: "-e -V -B"
+#              condition: and(succeeded(), eq(variables['main_build'], 'true'))
```
Comment (ppatierno) [inline .azure/templates/jobs/build_docs.yaml:30]: this commented part has to be here for any future reference or could be deleted?

**Review by ppatierno — APPROVED** [review]
LGTM. Just a question about a commented YAML.

File: development-docs/RELEASING.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+Review and commit the changes done by the `make` command and push them into the repository.
+The build pipeline should automatically start for any new commit pushed into the release branch.
+
+### Running the release pipeline
+
+Wait until the build pipeline is (successfully) finished for the last commit in the release branch.
+The regular pipeline will build the release ZIP / TAR.GZ files as well as the documentation and store them as artifacts.
+It will also stage the Maven artifacts.
+Then run the release pipeline manually from the Azure Pipelines UI.
+The release pipeline is names `kafka-bridge-release`.
```
Comment (Frawless) [inline development-docs/RELEASING.md:34]:
```suggestion
The release pipeline is named `kafka-bridge-release`.
```

**Review by Frawless — APPROVED** [review]

**Pull request merged**